### PR TITLE
New Python Cheat Sheet

### PR DESCRIPTION
**Pull request by ethanchewy**
## Demo

Demo GIF

See Video Here: https://drive.google.com/file/d/0B_0VuTZLjvTQVDlSamczSHgyTDg/view?usp=sharing
**What does your Instant Answer do?**
Gives a quick reference to Python Methods and Constants. Its a pretty condensed list. 

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It saves python developers time so that they can quickly reference important and most used Python Methods and Constants. This instant answer saves them time from switching links. 

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://overapi.com/python/ which gets its info from https://docs.python.org/. 

**Why did you choose this data source?**
It was the most in depth data source. I myself eliminated complicated language and condensed it succinctly.

**Are there any other alternative (better) data sources?**
Not really, I drew inspiration from stackoverflow, http://www.tutorialspoint.com/, and pythontutorial to make sure my instant answer only presented the most used/ most important methods and constants.

**What are some example queries that trigger this Instant Answer?**
python, py

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Python developers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes, IamRafy was working on this a couple weeks ago. I contacted him but he has not replied yet. Hopefully I can help him out with this pull request.

**Which existing Instant Answers will this one supercede/overlap with?**
#1622

**Are you having any problems? Do you need our help with anything?**
No

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

## Checklist
Please place an 'X' where appropriate.

- [X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 10
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] IE 11
        - [X] Microsoft edge
        
    - Windows 8
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] IE 10

    - Windows 7
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] IE 8
        - [X] IE 9
        - [X] IE 10

    - Windows XP
        - [X] IE 7
        - [X] IE 8

    - Linux
        - [X] Google Chrome
        - [X] Firefox
        
    - Mac OSX
        - [X] Google Chrome
        - [X] Firefox
        - [X] Opera
        - [X] Safari

    - iOS (iPhone)
        - [X] Safari Mobile
        - [X] Google Chrome
        - [X] Opera

    - iOS (iPad)
        - [X] Safari Mobile
        - [X] Google Chrome
        - [X] Opera

    - Android
        - [X] Firefox
        - [X] Native Browser
        - [X] Google Chrome
        - [X] Opera

-----

IA Page: https://duck.co/ia/view/python_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/python.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            "key": "upper()"  
+        }, {             
+            "val": "Returns a string in all lowercase",
+            "key": "lower()"
+        }, {
+            "val": "Returns a string with first character capitalized, the rest lower",
+            "key": "capitalize()"  
+        }, {
+            "val": "Returns a string with the leading and trailing whitespace removed",
+            "key": "strip ()"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/python.json]: nit: there's an extra space before the `()`

**Comment by moollaza** [comment]
@ethanchewy this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

By the way, we think you did a great job with this! Are you interested in helping others who build similar Instant Answers in the future?

If yes, we'll reach out and add you to a private list of mentors as well as send over some DDG merch as an additional thank you. Just email <open@duckduckgo.com> with the subject 'I want to help.' Please include your shipping address, phone-number (required for shipping outside USA) and t-shirt size. We'll reach out from there!

Otherwise, thank you for this amazing contribution. We hope you'll keep exploring and playing around with DuckDuckHack.

**Comment by ethanchewy** [comment]
Thanks for the kind words! I'm definitely interested in helping build similar Instant Answers (I always love helping out + merch is awesome)! Will send an email soon.

I'm currently exploring different forms of Instant Answers (besides cheat sheets) and going through those tutorials :) .

**Comment by ethanchewy** [comment]
@moollaza Also, do you mind listing me as a contributor for this cheatsheet? In beta, my name isn't listed for some reason.

**Comment by moollaza** [comment]
@ethanchewy congrats! It's live: https://beta.duckduckgo.com/?q=python+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by ethanchewy** [comment]
@moolaza thanks!

**Pull request merged**